tests/Aspire.Playground.Tests/Infrastructure/DistributedApplicationTestFactory.cs

```diff
@@ -47,4 +50,23 @@ public static async Task<IDistributedApplicationTestingBuilder> CreateAsync(stri
 
         return builder;
     }
+
+    internal sealed class ContainerRegistryHook : IDistributedApplicationLifecycleHook
```

Comment:
I think we should look at making the overload public instead of doing this.

Reply:
This can be removed if it is made public.

Reply:
I opened https://github.com/dotnet/aspire/issues/5285 to track that.

Reply:
Note that we only mirror dockerhub images on our Azure container registry. So we will need to know up front which apps will need this override, and which won't.

Where below we are checking `if (registry.Contains("docker.io")` to do the override.